Compilation error no member named 'baseid' during check of overlap.

While testing the new flag --trace-allocations I ran into this issue, which appears to be somewhat new (code compiles fine with older versions of pythran)

```
import numpy as np

# pythran export test()
def test():
    N = 100
    B = np.zeros((N,N), dtype=int)
    A = [[0] for ii in range(N)]
    for ii in range(len(A)):
        a = len(A[ii])
        B[ii, 0:a] = A[ii][0:a]
```
Here's some of the error message
```
/Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:59:14: error: no member named 'baseid' in '(anonymous namespace)::pythonic::types::sliced_list<long, (anonymous namespace)::pythonic::types::fast_contiguous_slice>'
    return e.baseid();
           ~ ^
/Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:65:12: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::baseid_helper<(anonymous namespace)::pythonic::types::sliced_list<long, (anonymous namespace)::pythonic::types::fast_contiguous_slice>>' requested here
    return baseid_helper(
           ^
/Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:72:26: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::baseid<(anonymous namespace)::pythonic::types::sliced_list<long, (anonymous namespace)::pythonic::types::fast_contiguous_slice>>' requested here
    return baseid(e0) == baseid(e1);
                         ^
/Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:456:9: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::may_overlap<(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<long, (anonymous namespace)::pythonic::types::pshape<std::integral_constant<long, 100>, std::integral_constant<long, 100>>> &>, (anonymous namespace)::pythonic::types::cstride_normalized_slice<1>>, (anonymous namespace)::pythonic::types::sliced_list<long, (anonymous namespace)::pythonic::types::fast_contiguous_slice>>' requested here
    if (may_overlap(*this, expr)) {
        ^
/Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:489:12: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<long, (anonymous namespace)::pythonic::types::pshape<std::integral_constant<long, 100>, std::integral_constant<long, 100>>> &>, (anonymous namespace)::pythonic::types::cstride_normalized_slice<1>>::_copy<(anonymous namespace)::pythonic::types::sliced_list<long, (anonymous namespace)::pythonic::types::fast_contiguous_slice>>' requested here
    return _copy(expr);
```

--- a/pythran/pythonic/include/types/list.hpp
+++ b/pythran/pythonic/include/types/list.hpp
@@ -178,6 +178,11 @@ namespace types
     bool contains(V const &v) const;
     intptr_t id() const;
 
+    intptr_t baseid() const
+    {
+      return reinterpret_cast<intptr_t>(&(*_data));
+    }
+
     long count(T const &x) const;
     template <class Tp, class Sp>
     friend std::ostream &operator<<(std::ostream &os,

--- a/pythran/tests/test_ndarray.py
+++ b/pythran/tests/test_ndarray.py
@@ -740,6 +740,21 @@ def newaxis8(n):
                       numpy.arange(4).reshape(2,2), numpy.array([[0, 1], [2, 3]]),
                       gexpr_baseid1=[NDArray[int,:,:], NDArray[int, :, :]])
 
+    def test_gexpr_baseid2(self):
+        code = """
+import numpy as np
+def gexpr_baseid2(N):
+    B = np.zeros((N,N), dtype=int)
+    A = [[0] for ii in range(N)]
+    for ii in range(len(A)):
+        a = len(A[ii])
+        B[ii, 0:a] = A[ii][0:a]
+    return B
+        """
+        self.run_test(code,
+                      100,
+                      gexpr_baseid2=[int])
+
     def test_gexpr_copy0(self):
         self.run_test("def gexpr_copy0(a,b): a[:,0] = b[:,0]; return a",
                       numpy.arange(16).reshape(8,2), numpy.arange(16).reshape(8,2),

